scripts/pylib/pytest-twister-harness/src/twister_harness/helpers/shell.py

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+import logging
+import re
+import time
+
+from twister_harness.device.device_adapter import DeviceAdapter
+from twister_harness.exceptions import TwisterHarnessTimeoutException
+
+logger = logging.getLogger(__name__)
+
+
+class Shell:
+    """
+    Helper class that provides methods used to interact with shell application.
+    """
+
+    def __init__(self, device: DeviceAdapter, prompt: str = 'uart:~$', timeout: float | None = None) -> None:
+        self._device: DeviceAdapter = device
+        self.prompt: str = prompt
+        self.base_timeout: float = timeout or device.base_timeout
+
+    def wait_for_prompt(self, timeout: float | None = None) -> bool:
+        """
+        Send every 0.5 second "enter" command to the device until shell prompt
+        statement will occur (return True) or timeout will be exceeded (return
+        False).
+        """
+        timeout = timeout or self.base_timeout
+        timeout_time = time.time() + timeout
+        self._device.clear_buffer()
+        while time.time() < timeout_time:
+            self._device.write(b'\n')
+            try:
+                line = self._device.readline(timeout=0.5, print_output=False)
+            except TwisterHarnessTimeoutException:
+                # ignore read timeout and try to send enter once again
+                continue
+            if self.prompt in line:
+                logger.debug('Got prompt')
+                time.sleep(0.05)
+                self._device.clear_buffer()
+                return True
+        return False
+
+    def exec_command(self, command: str, timeout: float | None = None) -> list[str]:
+        """
+        Send shell command to a device and return response. Passed command
+        is extended by double enter sings - first one to execute this command
+        on a device, second one to receive next prompt what is a signal that
+        execution was finished.
+        """
+        timeout = timeout or self.base_timeout
+        command = f'{command}\n\n'
+        self._device.clear_buffer()
+        self._device.write(command.encode())
+        lines: list[str] = self._device.readlines_until(regex=re.escape(self.prompt), timeout=timeout)
```

Comment:
first prompt (with a command) should be received immediately, so timeout can be set to 0.1

Reply:
What exactly means "immediately"? I can imagine that for some platforms this 0.1s could not be enough. `timeout` can be changed by users and if they don't want to wait too much for "no response" it can be set for some lower value (like 1 sec). At this moment I would like to stay with this version because it is safer IMO.

Reply:
"immadiately", just after pressing 'Enter' you must receive response ... ok, maybe 0.1s is to small, but 1sec should be enough.
I though about case, when there is some problem with the shell (not catched during initialization, e.g. promt message redefined). Than to fail test faster, do not stuck for 60s for every test.

Reply:
Was changed to 1.0 sec.